### ISSUE TITLE
Prefix search with multiple tokens crashes

### ISSUE THREAD

**Issue by zhlangsas** [RedisLabsModules/RediSearch#96]
When performing a prefix search with more than 1 token crashes redis. e.g. `FT.SEARCH myIdx "@name:(usa* bin*)"`

```
=== REDIS BUG REPORT START: Cut & paste starting from here ===
1:M 31 May 13:28:50.888 # Redis 999.999.999 crashed by signal: 11
1:M 31 May 13:28:50.889 # Crashed running the instuction at: 0x7fc72cd9ba19
1:M 31 May 13:28:50.889 # Accessing address: (nil)
1:M 31 May 13:28:50.889 # Failed assertion: <no assertion failed> (<no file>:0)
2017-05-31T13:28:50.889367774Z 
------ STACK TRACE ------
EIP:
/usr/src/RediSearch/src/redisearch.so(UI_SkipTo+0x159)[0x7fc72cd9ba19]
2017-05-31T13:28:50.889390574Z 
Backtrace:
redis-server *:6379(logStackTrace+0x29)[0x4679f9]
redis-server *:6379(sigsegvHandler+0xa6)[0x468096]
/lib/x86_64-linux-gnu/libpthread.so.0(+0xf890)[0x7fc72f1b4890]
/usr/src/RediSearch/src/redisearch.so(UI_SkipTo+0x159)[0x7fc72cd9ba19]
2017-05-31T13:28:50.889690174Z 
------ INFO OUTPUT ------
# Server
redis_version:999.999.999
redis_git_sha1:00000000
redis_git_dirty:0
redis_build_id:7086eaa1ec151505
redis_mode:standalone
os:Linux 4.9.13-moby x86_64
arch_bits:64
multiplexing_api:epoll
atomicvar_api:atomic-builtin
gcc_version:4.9.2
process_id:1
run_id:12a4399769c71c3c478de0ef3c36281f3f52e682
tcp_port:6379
uptime_in_seconds:25
uptime_in_days:0
hz:10
lru_clock:3065106
executable:/data/redis-server
config_file:
# Clients
connected_clients:1
client_longest_output_list:0
client_biggest_input_buf:0
blocked_clients:1
# Memory
used_memory:22475480
used_memory_human:21.43M
used_memory_rss:33787904
used_memory_rss_human:32.22M
used_memory_peak:22475480
used_memory_peak_human:21.43M
used_memory_peak_perc:100.23%
used_memory_overhead:3544350
used_memory_startup:765600
used_memory_dataset:18931130
used_memory_dataset_perc:87.20%
total_system_memory:2095898624
total_system_memory_human:1.95G
used_memory_lua:37888
used_memory_lua_human:37.00K
maxmemory:0
maxmemory_human:0B
maxmemory_policy:noeviction
mem_fragmentation_ratio:1.50
mem_allocator:jemalloc-4.0.3
active_defrag_running:0
lazyfree_pending_objects:0
# Persistence
loading:0
rdb_changes_since_last_save:0
rdb_bgsave_in_progress:0
rdb_last_save_time:1496237305
rdb_last_bgsave_status:ok
rdb_last_bgsave_time_sec:-1
rdb_current_bgsave_time_sec:-1
rdb_last_cow_size:0
aof_enabled:0
aof_rewrite_in_progress:0
aof_rewrite_scheduled:0
aof_last_rewrite_time_sec:-1
aof_current_rewrite_time_sec:-1
aof_last_bgrewrite_status:ok
aof_last_write_status:ok
aof_last_cow_size:0
# Stats
total_connections_received:1
total_commands_processed:3
instantaneous_ops_per_sec:0
total_net_input_bytes:169
total_net_output_bytes:10964
instantaneous_input_kbps:0.00
instantaneous_output_kbps:0.00
rejected_connections:0
sync_full:0
sync_partial_ok:0
sync_partial_err:0
expired_keys:0
evicted_keys:0
keyspace_hits:23
keyspace_misses:0
pubsub_channels:0
pubsub_patterns:0
latest_fork_usec:0
migrate_cached_sockets:0
slave_expires_tracked_keys:0
active_defrag_hits:0
active_defrag_misses:0
active_defrag_key_hits:0
active_defrag_key_misses:0
# Replication
role:master
connected_slaves:0
master_replid:6f8135f3983f7b614ba9e99bd6baacab7bfa7f06
master_replid2:0000000000000000000000000000000000000000
master_repl_offset:0
second_repl_offset:-1
repl_backlog_active:0
repl_backlog_size:1048576
repl_backlog_first_byte_offset:0
repl_backlog_histlen:0
# CPU
used_cpu_sys:0.08
used_cpu_user:0.19
used_cpu_sys_children:0.00
used_cpu_user_children:0.00
# Commandstats
cmdstat_command:calls=1,usec=306,usec_per_call=306.00
# Cluster
cluster_enabled:0
# Keyspace
db0:keys=55120,expires=0,avg_ttl=0
2017-05-31T13:28:50.890496074Z 
------ CLIENT LIST OUTPUT ------
id=2 addr=127.0.0.1:40054 fd=8 name= age=11 idle=0 flags=b db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=32768 obl=0 oll=0 omem=0 events=r cmd=FT.SEARCH
2017-05-31T13:28:50.890515374Z 
------ REGISTERS ------
1:M 31 May 13:28:50.889 # 
RAX:0000000000000002 RBX:0000000000000000
RCX:0000000000000004 RDX:0000000000000000
RDI:00007fc6f921e660 RSI:00007fc6ec000eb0
RBP:0000000000000028 RSP:00007fc72c564bf0
R8 :00007fc72cff8640 R9 :000000000000004d
R10:0000000000000020 R11:00007fc6f969dbe0
R12:00007fc6f922a380 R13:00007fc72c564c78
R14:0000000000000008 R15:00007fc6ec000e60
RIP:00007fc72cd9ba19 EFL:0000000000010202
CSGSFS:002b000000000033
1:M 31 May 13:28:50.889 # (00007fc72c564bff) -> 00007fc72c564cf8
1:M 31 May 13:28:50.889 # (00007fc72c564bfe) -> 00007fc6ec0009f0
1:M 31 May 13:28:50.889 # (00007fc72c564bfd) -> 00007fc72cd9bc03
1:M 31 May 13:28:50.889 # (00007fc72c564bfc) -> 0000000000000004
1:M 31 May 13:28:50.889 # (00007fc72c564bfb) -> 0000000000000001
1:M 31 May 13:28:50.889 # (00007fc72c564bfa) -> 00007fc72c564c78
1:M 31 May 13:28:50.889 # (00007fc72c564bf9) -> 00007fc6ec000ed0
1:M 31 May 13:28:50.889 # (00007fc72c564bf8) -> 0000000000000001
1:M 31 May 13:28:50.889 # (00007fc72c564bf7) -> 00007fc6ec000b20
1:M 31 May 13:28:50.889 # (00007fc72c564bf6) -> 0000000000000000
1:M 31 May 13:28:50.889 # (00007fc72c564bf5) -> 0000000000000000
1:M 31 May 13:28:50.889 # (00007fc72c564bf4) -> 0000000000000000
1:M 31 May 13:28:50.889 # (00007fc72c564bf3) -> 00007fc72c564c18
1:M 31 May 13:28:50.889 # (00007fc72c564bf2) -> 0000059f00000001
1:M 31 May 13:28:50.889 # (00007fc72c564bf1) -> 0000000200000000
1:M 31 May 13:28:50.889 # (00007fc72c564bf0) -> 0000000000000014
2017-05-31T13:28:50.890761974Z 
------ FAST MEMORY TEST ------
1:M 31 May 13:28:50.889 # Bio thread for job type #0 terminated
1:M 31 May 13:28:50.889 # Bio thread for job type #1 terminated
1:M 31 May 13:28:50.889 # Bio thread for job type #2 terminated
*** Preparing to test memory region 751000 (94208 bytes)
*** Preparing to test memory region 14f6000 (6352896 bytes)
*** Preparing to test memory region 7fc6ec000000 (135168 bytes)
*** Preparing to test memory region 7fc6f4000000 (135168 bytes)
*** Preparing to test memory region 7fc6f9200000 (27262976 bytes)
*** Preparing to test memory region 7fc6fad04000 (8388608 bytes)
*** Preparing to test memory region 7fc6fb505000 (8388608 bytes)
*** Preparing to test memory region 7fc6fbd06000 (8388608 bytes)
*** Preparing to test memory region 7fc6fc507000 (8388608 bytes)
*** Preparing to test memory region 7fc6fcd08000 (8388608 bytes)
*** Preparing to test memory region 7fc6fd509000 (8388608 bytes)
*** Preparing to test memory region 7fc6fdd0a000 (8388608 bytes)
*** Preparing to test memory region 7fc6fe50b000 (8388608 bytes)
*** Preparing to test memory region 7fc6fed0c000 (8388608 bytes)
*** Preparing to test memory region 7fc6ff50d000 (8388608 bytes)
*** Preparing to test memory region 7fc6ffd0e000 (8388608 bytes)
*** Preparing to test memory region 7fc70050f000 (8388608 bytes)
*** Preparing to test memory region 7fc700d10000 (8388608 bytes)
*** Preparing to test memory region 7fc701511000 (8388608 bytes)
*** Preparing to test memory region 7fc701d12000 (8388608 bytes)
*** Preparing to test memory region 7fc702513000 (8388608 bytes)
*** Preparing to test memory region 7fc702d14000 (8388608 bytes)
*** Preparing to test memory region 7fc703515000 (8388608 bytes)
*** Preparing to test memory region 7fc703d16000 (8388608 bytes)
*** Preparing to test memory region 7fc704517000 (8388608 bytes)
*** Preparing to test memory region 7fc704d18000 (8388608 bytes)
*** Preparing to test memory region 7fc705519000 (8388608 bytes)
*** Preparing to test memory region 7fc705d1a000 (8388608 bytes)
*** Preparing to test memory region 7fc70651b000 (8388608 bytes)
*** Preparing to test memory region 7fc706d1c000 (8388608 bytes)
*** Preparing to test memory region 7fc70751d000 (8388608 bytes)
*** Preparing to test memory region 7fc707d1e000 (8388608 bytes)
*** Preparing to test memory region 7fc70851f000 (8388608 bytes)
*** Preparing to test memory region 7fc708d20000 (8388608 bytes)
*** Preparing to test memory region 7fc709521000 (8388608 bytes)
*** Preparing to test memory region 7fc709d22000 (8388608 bytes)
*** Preparing to test memory region 7fc70a523000 (8388608 bytes)
*** Preparing to test memory region 7fc70ad24000 (8388608 bytes)
*** Preparing to test memory region 7fc70b525000 (8388608 bytes)
*** Preparing to test memory region 7fc70bd26000 (8388608 bytes)
*** Preparing to test memory region 7fc70c527000 (8388608 bytes)
*** Preparing to test memory region 7fc70cd28000 (8388608 bytes)
*** Preparing to test memory region 7fc70d529000 (8388608 bytes)
*** Preparing to test memory region 7fc70dd2a000 (8388608 bytes)
*** Preparing to test memory region 7fc70e52b000 (8388608 bytes)
*** Preparing to test memory region 7fc70ed2c000 (8388608 bytes)
*** Preparing to test memory region 7fc70f52d000 (8388608 bytes)
*** Preparing to test memory region 7fc70fd2e000 (8388608 bytes)
*** Preparing to test memory region 7fc71052f000 (8388608 bytes)
*** Preparing to test memory region 7fc710d30000 (8388608 bytes)
*** Preparing to test memory region 7fc711531000 (8388608 bytes)
*** Preparing to test memory region 7fc711d32000 (8388608 bytes)
*** Preparing to test memory region 7fc712533000 (8388608 bytes)
*** Preparing to test memory region 7fc712d34000 (8388608 bytes)
*** Preparing to test memory region 7fc713535000 (8388608 bytes)
*** Preparing to test memory region 7fc713d36000 (8388608 bytes)
*** Preparing to test memory region 7fc714537000 (8388608 bytes)
*** Preparing to test memory region 7fc714d38000 (8388608 bytes)
*** Preparing to test memory region 7fc715539000 (8388608 bytes)
*** Preparing to test memory region 7fc715d3a000 (8388608 bytes)
*** Preparing to test memory region 7fc71653b000 (8388608 bytes)
*** Preparing to test memory region 7fc716d3c000 (8388608 bytes)
*** Preparing to test memory region 7fc71753d000 (8388608 bytes)
*** Preparing to test memory region 7fc717d3e000 (8388608 bytes)
*** Preparing to test memory region 7fc71853f000 (8388608 bytes)
*** Preparing to test memory region 7fc718d40000 (8388608 bytes)
*** Preparing to test memory region 7fc719541000 (8388608 bytes)
*** Preparing to test memory region 7fc719d42000 (8388608 bytes)
*** Preparing to test memory region 7fc71a543000 (8388608 bytes)
*** Preparing to test memory region 7fc71ad44000 (8388608 bytes)
*** Preparing to test memory region 7fc71b545000 (8388608 bytes)
*** Preparing to test memory region 7fc71bd46000 (8388608 bytes)
*** Preparing to test memory region 7fc71c547000 (8388608 bytes)
*** Preparing to test memory region 7fc71cd48000 (8388608 bytes)
*** Preparing to test memory region 7fc71d549000 (8388608 bytes)
*** Preparing to test memory region 7fc71dd4a000 (8388608 bytes)
*** Preparing to test memory region 7fc71e54b000 (8388608 bytes)
*** Preparing to test memory region 7fc71ed4c000 (8388608 bytes)
*** Preparing to test memory region 7fc71f54d000 (8388608 bytes)
*** Preparing to test memory region 7fc71fd4e000 (8388608 bytes)
*** Preparing to test memory region 7fc72054f000 (8388608 bytes)
*** Preparing to test memory region 7fc720d50000 (8388608 bytes)
*** Preparing to test memory region 7fc721551000 (8388608 bytes)
*** Preparing to test memory region 7fc721d52000 (8388608 bytes)
*** Preparing to test memory region 7fc722553000 (8388608 bytes)
*** Preparing to test memory region 7fc722d54000 (8388608 bytes)
*** Preparing to test memory region 7fc723555000 (8388608 bytes)
*** Preparing to test memory region 7fc723d56000 (8388608 bytes)
*** Preparing to test memory region 7fc724557000 (8388608 bytes)
*** Preparing to test memory region 7fc724d58000 (8388608 bytes)
*** Preparing to test memory region 7fc725559000 (8388608 bytes)
*** Preparing to test memory region 7fc725d5a000 (8388608 bytes)
*** Preparing to test memory region 7fc72655b000 (8388608 bytes)
*** Preparing to test memory region 7fc726d5c000 (8388608 bytes)
*** Preparing to test memory region 7fc72755d000 (8388608 bytes)
*** Preparing to test memory region 7fc727d5e000 (8388608 bytes)
*** Preparing to test memory region 7fc72855f000 (8388608 bytes)
*** Preparing to test memory region 7fc728d60000 (8388608 bytes)
*** Preparing to test memory region 7fc729561000 (8388608 bytes)
*** Preparing to test memory region 7fc729d62000 (8388608 bytes)
*** Preparing to test memory region 7fc72a563000 (8388608 bytes)
*** Preparing to test memory region 7fc72ad64000 (8388608 bytes)
*** Preparing to test memory region 7fc72b565000 (8388608 bytes)
*** Preparing to test memory region 7fc72bd66000 (8388608 bytes)
*** Preparing to test memory region 7fc72c567000 (8388608 bytes)
*** Preparing to test memory region 7fc72cffe000 (8388608 bytes)
*** Preparing to test memory region 7fc72d7ff000 (8388608 bytes)
*** Preparing to test memory region 7fc72e000000 (12582912 bytes)
*** Preparing to test memory region 7fc72f1a1000 (16384 bytes)
*** Preparing to test memory region 7fc72f3be000 (16384 bytes)
*** Preparing to test memory region 7fc72fadd000 (16384 bytes)
*** Preparing to test memory region 7fc72fae3000 (16384 bytes)
*** Preparing to test memory region 7fc72fae9000 (4096 bytes)
.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.
```

**Comment by dvirsky** [comment]
Fixed in current master, thanks for the great catch!

**Comment by dvirsky** [comment]
Re-released 0.19 with this bug fix (And another minor one)
